frontend/src/components/experimentRuns/ExperimentRunValidation.tsx

```diff
@@ -149,7 +150,15 @@ interface LanePanelProps {
 
 function LanePanel({ lane, canValidate, canReset, isValidationInProgress, setPassed, setFailed, setAvailable }: LanePanelProps) {
 	const dispatch = useAppDispatch()
-  const datasetsById = useAppSelector(selectDatasetsByID)
+  const datasetsById = useAppSelector((state) => lane.datasets.map((dataset) => {
+    const datasetSelector = selectDatasetsByID(state)[dataset.datasetID]
+    return datasetSelector}).reduce((selectors, dataset) => {
+      if (dataset) {
+        selectors[dataset.id] = dataset;
+      }
+      return selectors;  
+    }, {}), shallowEqual
+  )
```

Comment:
This can be split up into `useAppSelector` (for `selectDatasetsByID(state)[dataset.datasetID]`) and `useMemo` (for creating `datasetsById`)

What a minute... `selectDatasetsByID` already returns datasetsByID. What is going on...

Reply:
Oh I see, you just want a subset of datasets.

Reply:
Some code formatting would help.